### PR TITLE
Run team code

### PR DESCRIPTION
**Pull request by dhood**
This sets up a container that runs a competitor's code, and configures a docker network so it can talk to the ARIAC server.

Something that I would have preferred is to [run the user scripts](https://github.com/osrf/ariac-docker/compare/run_team_code?expand=1#diff-fef727b32591fdb51d12bbb67102723dR38) as the `ariac-user` user, instead of root. I would prefer that because then it is more like the typical terminal experience for users. But, then they get prompted for passwords when installing things, e.g. [at this line](https://github.com/osrf/ariac-docker/compare/run_team_code?expand=1#diff-42141914b27260099c286b4c76463bcfR8) which I didn't spend enough time on to find a work around.

### CODE REVIEW

**Comment by j-rivero** [comment]
> But, then they get prompted for passwords when installing things, e.g. at this line which I didn't spend enough time on to find a work around.

Looks like we are install `wget` but, if I'm not wrong, the system should have `wget` already installed. They instruction [is in the Dockerfile](https://github.com/osrf/ariac-docker/compare/run_team_code?expand=1#diff-fef727b32591fdb51d12bbb67102723dR10).

We probably also need sudo to run `rosdep init`. Same solution than before, run in the `Dockefile`. The [official ROS docker images are using it](https://github.com/osrf/docker_images/blob/d9efe4367b9d376fd97a154bdaba896ca0382645/ros/indigo/indigo-ros-core/Dockerfile#L23)

**Comment by dhood** [comment]
> Looks like we are install wget but, if I'm not wrong, the system should have wget already installed. They instruction is in the Dockerfile.

I added that afterwards to install ariac. Yes, the wget in the script is redundant at the moment, but the key point is that users should be able to install something (wget or whatever else) from their installation script without dealing with the password prompt. My solution so far has been to run the script as root but it would be closer to user experience if it were run as a normal user with sudo priviledges.

> We probably also need sudo to run rosdep init. Same solution than before, run in the Dockefile. The official ROS docker images are using it

It outputs this warning so even that might be better as the ariac-user user
```
Step 6/13 : RUN rosdep init     && rosdep update
 ---> Running in 4c6d4849f778
Wrote /etc/ros/rosdep/sources.list.d/20-default.list
Recommended: please run

	rosdep update

Warning: running 'rosdep update' as root is not recommended.
  You should run 'sudo rosdep fix-permissions' and invoke 'rosdep update' again without sudo.
```

File: user-docker/run_competitor_container.bash [hunk 1]
```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env bash
+
+CONTAINER="competitor-container"
```
Comment (j-rivero) [inline user-docker/run_competitor_container.bash]: I think that we need to change container and image name here:

```diff
index 0c77bb2..de7d192 100755
--- a/user-docker/run_competitor_container.bash
+++ b/user-docker/run_competitor_container.bash
@@ -1,7 +1,7 @@
 #!/usr/bin/env bash
 
-CONTAINER="competitor-container"
-IMAGE_NAME="user-ariac"
+CONTAINER="user-ariac"
+IMAGE_NAME="competitor-container"
 DOCKER_EXTRA_ARGS=""
 COMMAND=$1
 ```

**Review by j-rivero — COMMENTED** [review]
A part from the change in the container/image names the rest of it seems to be working on my system. We should be ready to approve and improve things in new PRs.

It would be good if we create the user documentation and give it another test since I found some usual actions (like stop/start the server) that will probably need help from docs.

Running `./run_competitor_container.bash` gives no feedback in the command line and it took me a while to see that things were moving in the simulator and after 30secs I received a console output (Joint States). As a minor thing provide a quick feedback on the console could help.

**Comment by j-rivero** [comment]
> My solution so far has been to run the script as root but it would be closer to user experience if it were run as a normal user with sudo privileges.

This all makes sense to me. We can merge it in the current status and leave the change as an improvement or do you want to get it done before merging?

> It outputs this warning so even that might be better as the ariac-user user

Oh yes, we can run `rosdep init` as root and `rosdep update` as the ariac user.

File: user-docker/run_competitor_container.bash (hunk 1 above)
Comment (j-rivero) [reply user-docker/run_competitor_container.bash]: In relation with this, it is weird that we have a docker image named with the postfix `-container`. It comes [from here](https://github.com/osrf/ariac-docker/pull/1/files#diff-5dc676f8954036c870eeb5f2425f757cR9). It should be easier if we name the docker images and containers something like:

Images: ariac-server (like now) and ariac-user
Containers: ariac-server-system (now is just test) and ariac-user-system

or any other combination with some consistence in naming.

Comment (dhood) [reply user-docker/run_competitor_container.bash]: thanks! totally right, I did some last minute renaming... fixed in https://github.com/osrf/ariac-docker/pull/1/commits/b91d0dba49210be7a417fdf2bb28415a513d420f

**Comment by dhood** [comment]
> This all makes sense to me. We can merge it in the current status and leave the change as an improvement or do you want to get it done before merging?

Let's merge this and then we can iterate :) thanks for looking it over

**Pull request merged**